### PR TITLE
mam2: channels serialization

### PR DESCRIPTION
**Pull request by thibault-martinez**
# Test Plan:
Added tests

### CODE REVIEW

File: mam/v2/mam/channel.c [hunk 1]
```diff
@@ -91,41 +91,98 @@ retcode_t mam_channel_unwrap(spongos_t *const spongos, trits_t *const buffer,
 }
 
 size_t mam_channel_serialized_size(mam_channel_t const *const channel) {
-  return 0;
+  size_t mss_size = mss_stored_size(&channel->mss);
+  size_t endpoints_size = mam_endpoints_serialized_size(channel->endpoints);
+
+  return pb3_sizeof_ntrytes(MAM2_CHANNEL_ID_SIZE / 3) +       // id
```
Comment (tsvisabo) [inline mam/v2/mam/channel.c]: NUMBER_OF_TRITS_IN_TRYTE

File: mam/v2/mam/channel.c [hunk 2]
```diff
@@ -91,41 +91,98 @@ retcode_t mam_channel_unwrap(spongos_t *const spongos, trits_t *const buffer,
...
+
+  if ((ret = pb3_decode_size_t(&size, buffer)) != RC_OK) {  // mss size
+    return ret;
+  }
+
+  MAM2_TRITS_DEF0(ts, MAM2_MSS_SKN_SIZE);
+  ts = MAM2_TRITS_INIT(ts, MAM2_MSS_SKN_SIZE);
+  trits_set_zero(ts);
+
+  trits_copy(trits_take(*buffer, 4), trits_take(ts, 4));
```
Comment (tsvisabo) [inline mam/v2/mam/channel.c:167]: We should have a Defined value for that skn part (height)

File: mam/v2/mam/channel.c [hunk 3]
```diff
@@ -17,7 +17,7 @@ trits_t mam_channel_id(mam_channel_t const *const channel) {
 }
 
 trits_t mam_channel_name(mam_channel_t const *const channel) {
-  return channel->mss.nonce1;
+  return channel->name;
```
Comment (tsvisabo) [inline mam/v2/mam/channel.c:21]: Nice

**Review by tsvisabo — DISMISSED** [review]

File: mam/v2/mam/channel.c (hunk 2 above)
Comment (thibault-martinez) [reply mam/v2/mam/channel.c:167]: Sure

**Review by tsvisabo — DISMISSED** [review]

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**